sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -586,4 +589,25 @@ public static Optional<SootMethod> findConcreteMethod(
             + " and in its superclasses and interfaces");
     return Optional.empty();
   }
+
+  /**
+   * @return the bound function, see @setBoundFunction
+   */
+  public BiFunction<SootMethod, InvokableStmt, Boolean> getBoundFunction()
+  {
+      return boundFunction;
+  }
+
+  /** Set a bound function to prune the call graph. The bound function accepts the source method and
+   * an InvokableStatement, which is in the source method's body, and determines if the call from
+   * the source method to the invocation target shall be added into the call graph.
+   * 
+   * This allows building an pruned & potentially incomplete call graph, with lower memory / CPU
+   * cost.
+   * 
+   * @param boundFunction */
+  public void setBoundFunction(BiFunction<SootMethod, InvokableStmt, Boolean> boundFunction)
+  {
+      this.boundFunction = boundFunction;
+  }
```

Comment:
lets set the function in the constructor only

Reply:
hmm.... but this would force all sub-classes implement two constructors - one with a bound function, one without (so to use the always-true function)..... instead of having this feature automatically

let's keep it this way, or change setter to a "with" method (fluent API)?

Reply:
if you are concerned bound function is set after constructCompleteCallGraph() is called, I can add a checker and throw an exception if user do that. thought?

Reply:
lets implement it as a method directly without BiPredicate etc. - that way someone can override it the same way its done with a linkedhashmap to transform it to a LRUcache via overriding removeoldestentry() - e.g. see https://ashcode.medium.com/lru-cache-using-linkedhashmap-java-85e78ee1b976

Reply:
I would still recommend keeping it this way. This is more a setting of the analyser than an extension to it. In general I would try to avoid overriding a non-abstract method, it's kind of breaking enclosure. In general composition is preferred than inheritance, see https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
I think if you configure it in the constructor, the developer might be more aware of the feature.